### PR TITLE
Add cull opacity perf test to device lab

### PR DESCRIPTION
**Pull request by liyuqian**
For https://github.com/flutter/flutter/issues/24712

This test verifies that https://github.com/flutter/engine/pull/6923 will speedup the average rasterize time of this test from ~150ms to ~10ms

Please see non-auto-generated files in https://github.com/flutter/flutter/pull/25381/commits/37b21d9fb4756a8d8d12ae0def1d0640740c8818

### CODE REVIEW

**Comment by liyuqian** [comment]
> @yjbanov @cbracken @Hixie :
> 
> Please ignore the auto-generated files (any file that's not .dart, .yaml, or .md is auto-generated).

I've managed to put all non-auto-generated files in https://github.com/flutter/flutter/pull/25381/commits/37b21d9fb4756a8d8d12ae0def1d0640740c8818 for an easier review.

File: dev/benchmarks/macrobenchmarks/test_driver/cull_opacity_perf.dart [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
```
Comment (cbracken) [inline dev/benchmarks/macrobenchmarks/test_driver/cull_opacity_perf.dart]: Here and two more below: 2018

Comment (liyuqian) [reply dev/benchmarks/macrobenchmarks/test_driver/cull_opacity_perf.dart]: Done.

File: dev/benchmarks/macrobenchmarks/lib/src/cull_opacity.dart [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Stack(children: List<Widget>.generate(50, (int i) => Positioned(
+      left: 0,
+      top: (200 * i).toDouble() + _offsetY.value,
+      child: Opacity(
+        opacity: 0.5,
+        child: RepaintBoundary(
+          child: Container(
+            // slightly change width to invalidate raster cache
```
Comment (cbracken) [inline dev/benchmarks/macrobenchmarks/lib/src/cull_opacity.dart]: Uber-nit: `// Slightly change width to invalidate raster cache.`

Initial cap + period at the end.

File: dev/benchmarks/macrobenchmarks/lib/src/cull_opacity.dart [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  Animation<double> _offsetY;
+  AnimationController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+
+    _controller = AnimationController(vsync: this, duration: Duration(seconds: 2));
+    _offsetY = Tween<double>(begin: 0, end: -1000.0).animate(_controller)..addListener((){
+      setState((){});
```
Comment (cbracken) [inline dev/benchmarks/macrobenchmarks/lib/src/cull_opacity.dart]: Nit: `() {}` (add a space between the two pairs of brackets)

File: dev/benchmarks/macrobenchmarks/lib/main.dart [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            child: const Text('Cull opacity'),
+            onPressed: (){
+              Navigator.pushNamed(context, kCullOpacityRouteName);
+            },
+          )
+        ],
+      ),
+    );
+  }
+}
```
Comment (cbracken) [inline dev/benchmarks/macrobenchmarks/lib/main.dart]: Nit: add trailing newline.

File: dev/benchmarks/macrobenchmarks/lib/main.dart [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      initialRoute: '/',
+      routes: <String, WidgetBuilder>{
+        '/': (BuildContext context) => HomePage(),
+        kCullOpacityRouteName: (BuildContext context) => CullOpacityPage(),
+      },
+    );
+  }
+}
+
+
```
Comment (cbracken) [inline dev/benchmarks/macrobenchmarks/lib/main.dart]: Just one blank line.

File: dev/benchmarks/macrobenchmarks/android/app/build.gradle [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+android {
+    compileSdkVersion 27
+
+    lintOptions {
+        disable 'InvalidPackage'
+    }
+
+    defaultConfig {
+        // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
```
Comment (cbracken) [inline dev/benchmarks/macrobenchmarks/android/app/build.gradle]: Delete this TODO (it'll affect the technical debt benchmark)

File: dev/benchmarks/macrobenchmarks/android/app/build.gradle [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        targetSdkVersion 27
+        versionCode flutterVersionCode.toInteger()
+        versionName flutterVersionName
+        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+    }
+
+    buildTypes {
+        release {
+            // TODO: Add your own signing config for the release build.
+            // Signing with the debug keys for now, so `flutter run --release` works.
```
Comment (cbracken) [inline dev/benchmarks/macrobenchmarks/android/app/build.gradle]: Delete this TODO as well.

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
lgtm

File: dev/benchmarks/macrobenchmarks/lib/src/cull_opacity.dart (hunk 2 above)
Comment (liyuqian) [reply dev/benchmarks/macrobenchmarks/lib/src/cull_opacity.dart]: Done.

File: dev/benchmarks/macrobenchmarks/android/app/build.gradle (hunk 6 above)
Comment (liyuqian) [reply dev/benchmarks/macrobenchmarks/android/app/build.gradle]: Done.

File: dev/benchmarks/macrobenchmarks/android/app/build.gradle (hunk 7 above)
Comment (liyuqian) [reply dev/benchmarks/macrobenchmarks/android/app/build.gradle]: Done.

File: dev/benchmarks/macrobenchmarks/lib/main.dart (hunk 5 above)
Comment (liyuqian) [reply dev/benchmarks/macrobenchmarks/lib/main.dart]: Done.

File: dev/benchmarks/macrobenchmarks/lib/main.dart (hunk 4 above)
Comment (liyuqian) [reply dev/benchmarks/macrobenchmarks/lib/main.dart]: Done.

File: dev/benchmarks/macrobenchmarks/lib/src/cull_opacity.dart (hunk 3 above)
Comment (liyuqian) [reply dev/benchmarks/macrobenchmarks/lib/src/cull_opacity.dart]: Done.

**Review by liyuqian — COMMENTED** [review]
Thanks for the quick review!

**Pull request merged**